### ISSUE TITLE
clang compilation errors on macOS

### ISSUE THREAD

**Issue by philpennock** [vdukhovni/ssl_dane#1]
The -Werror promotes the warning to an error.

1.0.2k:
```
% make OPENSSL=/usr/local/opt/openssl 
cc -I/usr/local/opt/openssl/include -fPIC -Wall -Werror -g   -c -o danessl.o danessl.c
danessl.c:620:7: error: logical not is only applied to the left hand side of this comparison [-Werror,-Wlogical-not-parentheses]
                if (!X509_check_issued(ca, ca) == X509_V_OK) {
                    ^                          ~~
danessl.c:620:7: note: add parentheses after the '!' to evaluate the comparison first
                if (!X509_check_issued(ca, ca) == X509_V_OK) {
                    ^
                     (                                     )
danessl.c:620:7: note: add parentheses around left hand side expression to silence this warning
                if (!X509_check_issued(ca, ca) == X509_V_OK) {
                    ^
                    (                         )
1 error generated.
make: *** [danessl.o] Error 1
```

Same error with 1.1.0d via `make OPENSSL=/usr/local/opt/openssl@1.1`

Observed in Exim, traced down to import of this code.

**Comment by ErwanLegrand** [comment]
This issue can be closed. GitHub did not understand the commit message. "Fixes #1" would have worked.

**Comment by vdukhovni** [comment]
This is resolved.
